### DISCUSSION TITLE
Loading molecules from RDKit

### DISCUSSION THREAD

**Discussion by Tonylac77** [RMeli/spyrmsd#71] [Q&A] [answered]
Hi,

I am trying to figure out how to load molecules from RDKit. I would like to first load an SDF file using RDkit then run the RMSD calculations on that. Any tips or example code would be great.

Thanks

**Comment by RMeli** [comment, marked as answer]
Hi @Tonylac77, thank you for your interest in `spyrmsd`.

The `spyrmsd.molecules.Molecule` class has a special constructor to build a `spyrmsd` molecule directly from and RDKit molecule:

```python
from spyrmsd.molecule import Molecule
m = Molecule().from_rdkit(YOUR_RDKIT_MOLECULE)
```

You can then use [`spyrmsd.rmsd.rmsdwrapper`](https://spyrmsd.readthedocs.io/en/develop/api/spyrmsd.rmsd.html#spyrmsd.rmsd.rmsdwrapper) which works directly with `spyrmsd` molecules.

Alternatively, you can use the attributes of the `spyrmsd.molecule.Molecule` directly in the call to the more bare-bone [`spyrmsd.rmsd.symmrmsd`](https://spyrmsd.readthedocs.io/en/develop/api/spyrmsd.rmsd.html#spyrmsd.rmsd.symmrmsd), as shown in the [tutorial](https://spyrmsd.readthedocs.io/en/develop/tutorials/tutorial.html#symmetry-corrected-rmsd).

Finally, you can bypass the `spyrmsd.molecule.Molecule` class entirely by extracting the relevant information needed by `spyrmsd.rmsd.symmrmsd` directly from an RDKit molecule as follows (there might be better ways...):

https://github.com/RMeli/spyrmsd/blob/952e356ea8c2b1687aedbc43503f3459e0a7b21f/spyrmsd/optional/rdkit.py#L149-L165

---

```python
from rdkit import Chem
from rdkit.Chem import AllChem

# Instead of the following 4 lines of code, load your RDKit molecules from SDF files
rdmol1 = Chem.MolFromSmiles("c1ccccc1")
rdmol2 = Chem.MolFromSmiles("c1ccccc1")
AllChem.EmbedMolecule(rdmol1)
AllChem.EmbedMolecule(rdmol2)

from spyrmsd.molecule import Molecule
from spyrmsd.rmsd import rmsdwrapper

mol1 = Molecule.from_rdkit(rdmol1)
mol2 = Molecule.from_rdkit(rdmol2)

rmsdwrapper(mol1, mol2)
```

---

I hope this helps. I will try to improve the tutorial/documentation in this regard.
